### ISSUE TITLE
Does not compile on the latest nightly

### ISSUE THREAD

**Issue by euclio** [softprops/hubcaps#37]
```
┌ euclio@apollo ~/r/hubcaps (master)
└╌╌┄┄ ❯❯ rustc --version
rustc 1.10.0-nightly (764ef92ae 2016-05-19)
┌ euclio@apollo ~/r/hubcaps (master)
└╌╌┄┄ ❯❯ cargo test
   Compiling rustc-serialize v0.3.18
   Compiling gcc v0.3.26
   Compiling semver v0.1.20
   Compiling serde v0.7.0
   Compiling traitobject v0.0.1
   Compiling libc v0.2.8
   Compiling openssl v0.7.8
   Compiling language-tags v0.2.2
   Compiling memchr v0.1.10
   Compiling aho-corasick v0.5.1
   Compiling rand v0.3.14
   Compiling utf8-ranges v0.1.3
   Compiling lazy_static v0.1.15
   Compiling rustc_version v0.1.7
   Compiling unicase v1.4.0
   Compiling num_cpus v0.2.11
   Compiling num v0.1.31
   Compiling winapi-build v0.1.1
   Compiling typeable v0.1.2
   Compiling unicode-normalization v0.1.2
   Compiling serializable_enum v0.3.0
   Compiling bitflags v0.4.0
   Compiling kernel32-sys v0.2.1
   Compiling pkg-config v0.3.8
   Compiling matches v0.1.2
   Compiling unicode-bidi v0.2.3
   Compiling uuid v0.1.18
   Compiling openssl-sys v0.7.8
   Compiling openssl-sys-extras v0.7.8
   Compiling serde_json v0.7.0
   Compiling idna v0.1.0
   Compiling url v0.5.7
   Compiling httparse v1.1.1
   Compiling unicode-xid v0.0.3
   Compiling log v0.3.5
   Compiling url v1.1.0
   Compiling mime v0.2.0
   Compiling bitflags v0.5.0
   Compiling regex-syntax v0.3.0
   Compiling winapi v0.2.6
   Compiling hpack v0.2.0
   Compiling time v0.1.34
   Compiling term v0.2.14
   Compiling cookie v0.2.2
   Compiling solicit v0.4.4
   Compiling regex v0.1.58
   Compiling syntex_syntax v0.32.0
   Compiling syntex_syntax v0.30.0
   Compiling hyper v0.9.3
   Compiling env_logger v0.3.2
   Compiling syntex v0.32.0
   Compiling quasi v0.8.0
   Compiling aster v0.14.0
   Compiling syntex v0.30.0
   Compiling quasi_codegen v0.8.0
   Compiling serde_codegen v0.7.1
/home/euclio/repos/hubcaps/target/debug/build/serde_codegen-e34b6067af326668/out/lib.rs:15:5: 15:20 warning: attributes of the form `#[derive_*]` are reserved
for the compiler
/home/euclio/repos/hubcaps/target/debug/build/serde_codegen-e34b6067af326668/out/lib.rs:15     #[derive_Debug]
                                                                                               ^~~~~~~~~~~~~~~
/home/euclio/repos/hubcaps/target/debug/build/serde_codegen-e34b6067af326668/out/lib.rs:15:5: 15:20 help: add #![feature(custom_derive)] to the crate attribute
s to enable
/home/euclio/repos/hubcaps/target/debug/build/serde_codegen-e34b6067af326668/out/lib.rs:53:5: 53:20 warning: attributes of the form `#[derive_*]` are reserved
for the compiler
/home/euclio/repos/hubcaps/target/debug/build/serde_codegen-e34b6067af326668/out/lib.rs:53     #[derive_Debug]
                                                                                               ^~~~~~~~~~~~~~~
/home/euclio/repos/hubcaps/target/debug/build/serde_codegen-e34b6067af326668/out/lib.rs:53:5: 53:20 help: add #![feature(custom_derive)] to the crate attribute
s to enable
/home/euclio/repos/hubcaps/target/debug/build/serde_codegen-e34b6067af326668/out/lib.rs:104:5: 104:20 warning: attributes of the form `#[derive_*]` are reserve
d for the compiler
/home/euclio/repos/hubcaps/target/debug/build/serde_codegen-e34b6067af326668/out/lib.rs:104     #[derive_Debug]
                                                                                                ^~~~~~~~~~~~~~~
/home/euclio/repos/hubcaps/target/debug/build/serde_codegen-e34b6067af326668/out/lib.rs:104:5: 104:20 help: add #![feature(custom_derive)] to the crate attribu
tes to enable
/home/euclio/repos/hubcaps/target/debug/build/serde_codegen-e34b6067af326668/out/lib.rs:147:5: 147:20 warning: attributes of the form `#[derive_*]` are reserve
d for the compiler
/home/euclio/repos/hubcaps/target/debug/build/serde_codegen-e34b6067af326668/out/lib.rs:147     #[derive_Debug]
                                                                                                ^~~~~~~~~~~~~~~
/home/euclio/repos/hubcaps/target/debug/build/serde_codegen-e34b6067af326668/out/lib.rs:147:5: 147:20 help: add #![feature(custom_derive)] to the crate attribu
tes to enable
   Compiling hubcaps v0.2.1 (file:///home/euclio/repos/hubcaps)
build.rs:14:29: 14:42 error: mismatched types [E0308]
build.rs:14     serde_codegen::register(&mut registry);
                                        ^~~~~~~~~~~~~
build.rs:14:29: 14:42 help: run `rustc --explain E0308` to see a detailed explanation
build.rs:14:29: 14:42 note: expected type `&mut syntex::Registry`
build.rs:14:29: 14:42 note:    found type `&mut syntex::Registry`
build.rs:14:29: 14:42 note: Perhaps two different versions of crate `syntex` are being used?
build.rs:14     serde_codegen::register(&mut registry);
                                        ^~~~~~~~~~~~~
error: aborting due to previous error
error: Could not compile `hubcaps`.

To learn more, run the command again with --verbose.
```

**Comment by softprops** [comment]
I'll have a look. I believe this particular error 

```
build.rs:14:29: 14:42 note: expected type `&mut syntex::Registry`
build.rs:14:29: 14:42 note:    found type `&mut syntex::Registry`
```

happens when a down stream dependency needs sync'd with a new combination of its new downstream dependency versions. In this case it looks like the serde codegen crate.

**Comment by tafia** [comment]
I confirm the error, could you do a cargo update and publish a version with last syntex crate (0.32.0) in crates.io?

**Comment by softprops** [comment]
Ok published 0.2.2. nighly looks [green](https://travis-ci.org/softprops/hubcaps/jobs/134063384)
